include/secp256k1_schnorrsig.h

```diff
@@ -56,6 +56,85 @@ SECP256K1_API int secp256k1_schnorrsig_parse(
     const unsigned char *in64
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
+/** Anti Nonce Sidechannel Protocol
+ *
+ *  The next functions can be used to prevent a signing device from exfiltrating the secret signing
+ *  keys through biased signature nonces. The general idea is that a host provides additional
+ *  randomness to the signing device client and the client commits to the randomness in the nonce
+ *  using sign-to-contract.
+ *  In order to make the randomness unpredictable, the host and client must engage in a
+ *  commit-reveal protocol as follows:
+ *  1. The host draws the randomness, commits to it with the `anti_nonce_sidechan_host_commit`
+ *     function and sends the commitment to the client.
+ *  2. The client commits to its sign-to-contract original nonce (which is the nonce without the
+ *     sign-to-contract tweak) using the hosts commitment by calling the
+ *     `secp256k1_schnorrsig_anti_nonce_sidechan_client_commit` function. The client sends the
+ *     rusulting commitment to the host
+ *  3. The host replies with the randomness generated in step 1.
+ *  4. The client signs with `schnorrsig_sign` using the host provided randomness as `s2c_data` and
+ *     sends the signature and opening to the host.
+ *  5. The host checks that the signature contains an sign-to-contract commitment to the randomness
+ *     by calling `secp256k1_schnorrsig_anti_nonce_sidechan_host_verify` with the client's
+ *     commitment from step 2 and the signature and opening received in step 4. If verification does
+ *     not succeed, the protocol failed and can be restarted.
+ */
+
+/** Create a randomness commitment on the host as part of the Anti Nonce Sidechannel Protocol.
+ *
+ *  Returns 1 on success, 0 on failure.
+ *  Args:              ctx: pointer to a context object (cannot be NULL)
+ *  Out: rand_commitment32: pointer to 32-byte array to store the returned commitment (cannot be NULL)
+ *  In:             rand32: the 32-byte randomness to commit to (cannot be NULL)
```

Comment:
Maybe it's even better to make this function responsible for calling `secp256k1_rand256` to avoid that the user screws up when producing randomness. 

(But then the user can't use his fancy hardware RNG... If we want to support that, then maybe there could also a callback to a randomness function, and if it's NULL, then just call `lsecp256k1_rand256`. But in general, I don't like the idea of the user providing randomness, unless really necessary.)